src/main/sensors/barometer.c

```diff
@@ -174,11 +174,18 @@ int32_t baroCalculateAltitude(void)
 
 void performBaroCalibrationCycle(void)
 {
+    static int32_t savedGroundPressure=0;
+
     baroGroundPressure -= baroGroundPressure / 8;
     baroGroundPressure += baroPressureSum / PRESSURE_SAMPLE_COUNT;
     baroGroundAltitude = (1.0f - powf((baroGroundPressure / 8) / 101325.0f, 0.190295f)) * 4433000.0f;
 
-    calibratingB--;
+    if (baroGroundPressure==savedGroundPressure)
```

Comment:
Isn't this too fragile? I would expect noise much larger than single count. And two identical samples may stop calibration prematurely ...